### DISCUSSION TITLE
Overide existing type

### DISCUSSION THREAD

**Discussion by Romain-Devillez** [vendurehq/vendure#1870] [Q&A] [answered]
Hello there, 

I'm new on Vendure.

I would like to edit a type that is automatically generated.
I have searched the documentation, without success.

In my case, I would like to make all fields of "RegisterCustomerInput" mandatory

Thank you for your help.

![Capture d’écran 2022-11-10 à 17 13 50](https://user-images.githubusercontent.com/49993311/201148437-f647ceed-db27-4575-a31b-b440beeb0453.png)

![Capture d’écran 2022-11-10 à 17 13 58](https://user-images.githubusercontent.com/49993311/201148481-ee06e254-9524-40f8-ad75-f39bfb734274.png)

**Comment by michaelbromley** [comment, marked as answer]
Answered in Slack, but posting here so it is also public:

It is possible to extend existing types using the GraphQL extend keyword, as described [here](https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#add-fields-to-existing-types). However, it's not possible to change an existing field.

A solution to your use-case could be to [define a custom mutation](https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#adding-a-new-query-or-mutation), which allows you to specify exactly what the input should be. You can then [override the existing registerCustomerAccount](https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#override-built-in-resolvers) mutation with a custom resolver implementation that throws an error, so that people can only use your new custom mutation to register an account.
